### ISSUE TITLE
add a gitlabSourceRepoURL environment variable

### ISSUE THREAD

**Issue by klausbayrhammer** [jenkinsci/gitlab-plugin#92]
When jenkins builds are triggered by merge requests the `gitlabSourceRepoURL` parameter is passed to jenkins. Have you tought about passing a `gitlabTargetRepoURL` as well?

A `$gitlabTargetRepoURL` would make our CI quite a bit easier because we have multiple repositories which are built with the same build process - but as of yet we have to create separat jobs for each repository because the push-webhooks dont have the information about the repo which triggered the hook

**Comment by chrisferry** [comment]
:+1: We have over 30+ repositories that use the same identical job.  We need the ability to use a variable to set up the git repo URL.  Currently blocked on this feature

**Comment by jvdlaar** [comment]
We have the same situation, would be great to have such a variable.

**Comment by noelcapule** [comment]
yes, i agree. We have 100+ repos

**Comment by omehegan** [comment]
You have 1 build job/pipeline which can build numerous repos? That seems strange - then build status/history is basically worthless, isn't it?

1 job per repo seems to be a pretty widely accepted practice.

**Comment by klausbayrhammer** [comment]
> .. That seems strange - then build status/history is basically worthless, isn't it?

Yes but it doesn't matter because we report the build result to gitlabs merge request and in the end we only need the information there

> 1 job per repo seems to be a pretty widely accepted practice.

I'd say it depends on your setup. At least for us it works pretty well. Especially because I'm not aware of an elegant way to specify job-templates in jenkins without using the groovy-dsl

In addition to the requested `gitlabSourceRepoURL` I think it would be great to think about wrapping all parameters gitlab's merge/push reqests into parameters. (see https://gitlab.com/gitlab-org/gitlab-ce/blob/master/doc/web_hooks/web_hooks.md)

**Comment by coder-hugo** [comment]
Duplicate of #211
